auth/auth.js

```diff
@@ -1,159 +1,112 @@
-let jwt = require('jsonwebtoken');
+let jwt = require("jsonwebtoken");
 
-const { SECRET } = require('../config/config');
-const { SALT } = require('../config/config');
-const { clients } = require('../config/medicSoftDbMicroserviceURLs');
-const axios = require('axios');
-const bcrypt = require('bcryptjs');
-
-async function extractToken(req) {
-	let tokenHeaders = [ 'x-access-token', 'authorization' ];
-
-	let token = '';
-
-	for (let tokenHeader of tokenHeaders) {
-		if (tokenHeader in req.headers) {
-			token = req.headers[tokenHeader];
-		}
-	}
-
-	return token;
-}
-
-async function generateToken(email, clientdata) {
-	tokendata = {
-		email: email,
-		isDoctor: clientdata.isDoctor,
-		isActive: clientdata.isActive
-	};
-	return jwt.sign({ tokendata }, SECRET, {
-		expiresIn: '24h' /* expires in 24 hours */
-	});
-}
-
-async function generateUser(userDetails) {
-	if (validatePassword(userDetails)) {
-		userDetails.password = bcrypt.hashSync(userDetails.password, SALT);
-		let resultdata = await axios.post(clients.POST.clients, userDetails);
-		return resultdata;
-	}
-
-	function validatePassword(userDetails) {
-		if ('password' in userDetails) {
-			if (userDetails.password.length !== 0) {
-				return true;
-			} else {
-				throw new Error("Illegal arguments: password cannot be empty");
-			}
-		} else {
-			throw new Error("Illegal arguments: password is missing");
-		}
-	}
-}
-
-async function checkPassword(enteredPassword, storedPassword) {
-	let result = bcrypt.compareSync(enteredPassword, storedPassword);
-	return result;
-}
+const { SECRET } = require("../config/config");
+const utils = require("../utils/index");
 
 exports.testAuth = async (req, res, next) => {
-	try {
-		let token = await extractToken(req);
-
-		if (token) {
-			jwt.verify(token, SECRET, (err, decoded) => {
-				if (err) {
-					res.send({
-						success: false,
-						message: 'Token is not valid'
-					});
-				} else {
-					res.send({ decoded });
-					next();
-				}
-			});
-		} else {
-			res.send({
-				success: false,
-				message: 'Auth token is not supplied'
-			});
-		}
-	} catch (err) {
-		res.send({
-			success: false,
-			statusCode: 500,
-			statusText: 'Internal Server Error',
-			message: err.message
-		});
-	}
+  try {
+    let token = await utils.token.extractToken(req);
+
+    if (token) {
+      jwt.verify(token, SECRET, (err, decoded) => {
```

Comment:
I think, this token verification code, should also be moved to token.js

Reply:
You forgot to refactor this code part

Reply:
I am working on this.

Reply:
This is now complete